interfaces/builtin/maliit_input_method.go

```diff
@@ -0,0 +1,170 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const maliitInputMethodPermanentSlotAppArmor = `
+# Description: Allow operating as a maliit server.
+# Communication with maliit happens in the following stages:
+#  * An application connects to the address service: org.maliit.Server.Address.
+#  * The server responds with a private unix socket of the form
+#    @/tmp/maliit-server/dbus-* on which the server is running a peer-to-peer
+#    dbus session.
+#  * All further communication happens over this channel
+#  * An application wishing to receive input then requests that it be made the
+#    active context.
+#  * At this point maliit retrieves the application's PID based on the dbus
+#    channel and verifies with Unity 8 that the application is currently
+#    focused.
+#  * Only if the application is focused is it then able to receive input from
+#    the on-screen keyboard.
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+
+# allow binding to the maliit dbus server for address negotiation
+dbus (bind)
+    bus=session
+    name="org.maliit.server",
+
+# TODO: should this be somewhere else?
+/usr/share/glib-2.0/schemas/ r,
+
+# maliit uses peer-to-peer dbus over a unix socket after address negotiation.
+# Each application has its own one-to-one communication channel with the maliit
+# server, over which all further communication happens.
```

Comment:
Can you add to the end: `Send and receive rules are in the per-snap connection policy.`